src/ImageSharp.Drawing/Processing/Drawing/Processors/FillProcessor.cs

```diff
@@ -49,38 +50,57 @@ protected override void OnFrameApply(ImageFrame<TPixel> source, Rectangle source
             int minY = Math.Max(0, startY);
             int maxY = Math.Min(source.Height, endY);
 
-            // Reset offset if necessary.
-            if (minX > 0)
-            {
-                startX = 0;
-            }
-
-            if (minY > 0)
-            {
-                startY = 0;
-            }
-
             int width = maxX - minX;
 
-            using (IBuffer<float> amount = source.MemoryManager.Allocate<float>(width))
-            using (BrushApplicator<TPixel> applicator = this.brush.CreateApplicator(
-                source,
-                sourceRectangle,
-                this.options))
-            {
-                amount.Span.Fill(this.options.BlendPercentage);
+            var solidBrush = this.brush as SolidBrush<TPixel>;
 
+            // If there's no reason for blending, then avoid it.
+            if (solidBrush != null && this.options.BlendPercentage == 1f && solidBrush.Color.ToVector4().W == 1f)
```

Comment:
fast path should be activated when `solidBrush != null` and any of these things are true:

`this.options.BlenderMode == PixelBlenderMode.Normal && this.options.BlendPercentage == 1f && solidBrush.Color.ToVector4().W == 1f`
OR

`this.options.BlenderMode == PixelBlenderMode.Over && this.options.BlendPercentage == 1f && solidBrush.Color.ToVector4().W == 1f`

OR

`this.options.BlenderMode == PixelBlenderMode.Src`

Reply:
Changed it, thanks for the help here!